File: sdks/java/examples/BlockedDemo.java
import ai.varden.Varden;
import ai.varden.VardenBlockedException;

public final class BlockedDemo {
    public static void main(String[] args) throws Exception {
        Varden.protect();
        System.out.println("Varden Java demo: only Varden.protect() is required");
        try {
            Varden.command("python", "-c", "print('java demo')", "delete_database", "prod-customer-db").inheritIO().start();
            System.out.println("Expected block, but command ran");
        } catch (VardenBlockedException blocked) {
            System.out.println("Varden blocked the command as expected");
            System.out.println(blocked.result.decision.raw);
        }
    }
}

File: sdks/rust/examples/blocked_demo.rs
fn main() -> Result<(), Box<dyn std::error::Error>> {
    varden::protect()?;
    println!("Varden Rust demo: only varden::protect() is required");

    match varden::process::Command::new("python")?
        .arg("-c")
        .arg("print('rust demo')")
        .arg("delete_database")
        .arg("prod-customer-db")
        .spawn() {
        Ok(_) => println!("Expected a block, but process ran"),
        Err(varden::Error::Blocked(result)) => {
            println!("Varden blocked the process as expected");
            println!("{:?}", result.decision);
        }
        Err(other) => return Err(Box::new(other)),
    }

    Ok(())
}
